### PR TITLE
Adjust naming conventions and general refactoring in HLE Project

### PR DESCRIPTION
**Pull request by Thealexbarney**

### CODE REVIEW

File: Ryujinx.HLE/HOS/Diagnostics/Demangler/Ast/BinaryExpression.cs [hunk 1]
```diff
@@ -4,37 +4,37 @@ namespace Ryujinx.HLE.HOS.Diagnostics.Demangler.Ast
...
 
-        public BinaryExpression(BaseNode LeftPart, string Name, BaseNode RightPart) : base(NodeType.BinaryExpression)
+        public BinaryExpression(BaseNode leftPart, string name, BaseNode rightPart) : base(NodeType.BinaryExpression)
         {
-            this.LeftPart  = LeftPart;
-            this.Name      = Name;
-            this.RightPart = RightPart;
+            this._leftPart  = leftPart;
+            this._name      = name;
+            this._rightPart = rightPart;
```
Comment (emmauss) [inline Ryujinx.HLE/HOS/Diagnostics/Demangler/Ast/BinaryExpression.cs]: same

File: Ryujinx.HLE/HOS/Diagnostics/Demangler/Ast/BracedExpression.cs [hunk 2]
```diff
@@ -4,37 +4,37 @@ namespace Ryujinx.HLE.HOS.Diagnostics.Demangler.Ast
...
 
-        public BracedExpression(BaseNode Element, BaseNode Expression, bool IsArrayExpression) : base(NodeType.BracedExpression)
+        public BracedExpression(BaseNode element, BaseNode expression, bool isArrayExpression) : base(NodeType.BracedExpression)
         {
-            this.Element           = Element;
-            this.Expression        = Expression;
-            this.IsArrayExpression = IsArrayExpression;
+            this._element           = element;
+            this._expression        = expression;
+            this._isArrayExpression = isArrayExpression;
```
Comment (emmauss) [inline Ryujinx.HLE/HOS/Diagnostics/Demangler/Ast/BracedExpression.cs]: same

File: Ryujinx.HLE/HOS/Diagnostics/Demangler/Ast/BracedRangeExpression.cs [hunk 3]
```diff
@@ -4,31 +4,31 @@ namespace Ryujinx.HLE.HOS.Diagnostics.Demangler.Ast
...
 
-        public BracedRangeExpression(BaseNode FirstNode, BaseNode LastNode, BaseNode Expression) : base(NodeType.BracedRangeExpression)
+        public BracedRangeExpression(BaseNode firstNode, BaseNode lastNode, BaseNode expression) : base(NodeType.BracedRangeExpression)
         {
-            this.FirstNode  = FirstNode;
-            this.LastNode   = LastNode;
-            this.Expression = Expression;
+            this._firstNode  = firstNode;
+            this._lastNode   = lastNode;
+            this._expression = expression;
```
Comment (emmauss) [inline Ryujinx.HLE/HOS/Diagnostics/Demangler/Ast/BracedRangeExpression.cs]: same

File: Ryujinx.HLE/HOS/Diagnostics/Demangler/Ast/CastExpression.cs [hunk 4]
```diff
@@ -4,25 +4,25 @@ namespace Ryujinx.HLE.HOS.Diagnostics.Demangler.Ast
...
 
-        public CastExpression(string Kind, BaseNode To, BaseNode From) : base(NodeType.CastExpression)
+        public CastExpression(string kind, BaseNode to, BaseNode @from) : base(NodeType.CastExpression)
         {
-            this.Kind = Kind;
-            this.To   = To;
-            this.From = From;
+            this._kind = kind;
+            this._to   = to;
+            this._from = @from;
```
Comment (emmauss) [inline Ryujinx.HLE/HOS/Diagnostics/Demangler/Ast/CastExpression.cs]: just fix all the 'this'.

File: Ryujinx.HLE/HOS/Kernel/KThread.cs [hunk 5]
```diff
@@ -55,85 +55,85 @@ class KThread : KSynchronizationObject, IKFutureSchedulerObject
...
 
-            Context        = Thread;
-            Owner          = Process;
-            PreferredCore  = ProcessorId;
-            Scheduler      = System.Scheduler;
-            SchedulingData = System.Scheduler.SchedulingData;
+            Context        = thread;
+            Owner          = process;
+            PreferredCore  = processorId;
+            _scheduler      = system.Scheduler;
```
Comment (emmauss) [inline Ryujinx.HLE/HOS/Kernel/KThread.cs]: align

File: Ryujinx.HLE/HOS/Diagnostics/Demangler/Demangler.cs [hunk 6]
```diff
@@ -8,113 +8,113 @@ namespace Ryujinx.HLE.HOS.Diagnostics.Demangler
 {
     class Demangler
     {
-        private static readonly string BASE_36   = "0123456789abcdefghijklmnopqrstuvwxyz";
-        private List<BaseNode> SubstitutionList  = new List<BaseNode>();
-        private List<BaseNode> TemplateParamList = new List<BaseNode>();
+        private static readonly string Base36   = "0123456789abcdefghijklmnopqrstuvwxyz";
```
Comment (emmauss) [inline Ryujinx.HLE/HOS/Diagnostics/Demangler/Demangler.cs]: align

File: Ryujinx.HLE/HOS/Diagnostics/Demangler/Demangler.cs [hunk 7]
```diff
@@ -8,113 +8,113 @@ namespace Ryujinx.HLE.HOS.Diagnostics.Demangler
...
+        private bool _canParseTemplateArgs;
 
-        public Demangler(string Mangled)
+        public Demangler(string mangled)
         {
-            this.Mangled         = Mangled;
-            Position             = 0;
-            Length               = Mangled.Length;
-            CanParseTemplateArgs = true;
+            this.Mangled         = mangled;
```
Comment (emmauss) [inline Ryujinx.HLE/HOS/Diagnostics/Demangler/Demangler.cs]: align

File: Ryujinx.sln.DotSettings [hunk 8]
```diff
@@ -0,0 +1,20 @@
+﻿<wpf:ResourceDictionary xml:space="preserve" xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml" xmlns:s="clr-namespace:System;assembly=mscorlib" xmlns:ss="urn:shemas-jetbrains-com:settings-storage-xaml" xmlns:wpf="http://schemas.microsoft.com/winfx/2006/xaml/presentation">
```
[the remark this answers is not in the document]
Comment (Thealexbarney) [reply Ryujinx.sln.DotSettings]: It's used by Resharper. I added some of the code style settings used in the solution. It's more of a convenience for those using Resharper or Rider.

File: Ryujinx.HLE/FileSystem/Content/ContentManager.cs [hunk 9]
```diff
@@ -9,297 +9,298 @@ namespace Ryujinx.HLE.FileSystem.Content
...
-            SharedFontTitleDictionary = new Dictionary<string, long>()
+            _sharedFontTitleDictionary = new Dictionary<string, long>
             {
                 { "FontStandard",                  0x0100000000000811 },
                 { "FontChineseSimplified",         0x0100000000000814 },
                 { "FontExtendedChineseSimplified", 0x0100000000000814 },
                 { "FontKorean",                    0x0100000000000812 },
                 { "FontChineseTraditional",        0x0100000000000813 },
-                { "FontNintendoExtended" ,         0x0100000000000810 },
+                { "FontNintendoExtended" ,         0x0100000000000810 }
```
Comment (jduncanator) [inline Ryujinx.HLE/FileSystem/Content/ContentManager.cs]: There is an unneeded space after the string on this line.

File: Ryujinx.HLE/FileSystem/Content/ContentManager.cs [hunk 10]
```diff
@@ -9,297 +9,298 @@ namespace Ryujinx.HLE.FileSystem.Content
...
                 {
-                    ContentPathString   = LocationHelper.GetContentRoot(StorageId);
-                    ContentDirectory    = LocationHelper.GetRealPath(Device.FileSystem, ContentPathString);
-                    RegisteredDirectory = Path.Combine(ContentDirectory, "registered");
+                    contentPathString   = LocationHelper.GetContentRoot(storageId);
+                    contentDirectory    = LocationHelper.GetRealPath(_device.FileSystem, contentPathString);
+                    registeredDirectory = Path.Combine(contentDirectory, "registered");
                 }
-                catch (NotSupportedException NEx)
+                catch (NotSupportedException nEx)
```
Comment (jduncanator) [inline Ryujinx.HLE/FileSystem/Content/ContentManager.cs]: The exception variable can be completely removed here. Alternatively, it should probably be called nex not nEx.

File: Ryujinx.HLE/FileSystem/Content/ContentManager.cs [hunk 11]
```diff
@@ -9,297 +9,298 @@ namespace Ryujinx.HLE.FileSystem.Content
...
+        public bool HasNca(string ncaId, StorageId storageId)
         {
-            if (ContentDictionary.ContainsValue(NcaId))
+            if (_contentDictionary.ContainsValue(ncaId))
             {
-                var         Content     = ContentDictionary.FirstOrDefault(x => x.Value == NcaId);
-                long        TitleId     = (long)Content.Key.Item1;
-                ContentType ContentType = Content.Key.Item2;
-                StorageId   Storage     = GetInstalledStorage(TitleId, ContentType, StorageId);
+                KeyValuePair<(ulong, ContentType), string> content = _contentDictionary.FirstOrDefault(x => x.Value == ncaId);
```
Comment (jduncanator) [inline Ryujinx.HLE/FileSystem/Content/ContentManager.cs]: This should still use `var` in my opinion, as the type decl is overly complex.

**Comment by jduncanator** [comment]
I’ve left a few comments, I couldn’t get all the way through it as I’m on a phone at the moment, it it’s still not merged when I’m next at a computer, I’ll do more of a review.

**Comment by gdkchan** [comment]
This will need a rebase due to nvdec merge (sorry).

File: Ryujinx.HLE/Utilities/WSAError.cs [hunk 12]
```diff
@@ -1,82 +1,82 @@
 ﻿namespace Ryujinx.HLE.Utilities
 {
-    enum WSAError
+    internal enum WsaError
```
Comment (jduncanator) [inline Ryujinx.HLE/Utilities/WSAError.cs]: This enum maps external error codes internally, and should have their names left as is.

File: Ryujinx.HLE/Utilities/LinuxError.cs [hunk 13]
```diff
@@ -1,45 +1,45 @@
 namespace Ryujinx.HLE.Utilities
 {
-    enum LinuxError
+    internal enum LinuxError
```
Comment (jduncanator) [inline Ryujinx.HLE/Utilities/LinuxError.cs]: This enum maps external error codes internally, and should have their names left as is.

**Pull request merged**